themes/src/main/resources/theme/base/login/register-user-profile.ftl

```diff
@@ -49,6 +49,32 @@
 		            </#if>
                 </#if>  
             </@userProfileCommons.userProfileFormFields>
+
+            <#if termsAcceptanceRequired??>
```

Comment:
Would make sense to create a macro and import/re-use the template on both registration pages?